### PR TITLE
Use Swift Struct instead of String as the key in the cache.

### PR DESCRIPTION
**Pull request by mmadjer**
These changes are for #360.

### CODE REVIEW

File: IBAnimatable/PresentationPresenterChacheKey.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//  PresentationChacheKey.swift
+//  IBAnimatable
+//
+//  Created by Miro on 8/12/16.
+//  Copyright © 2016 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+
+struct PresentationPresenterChacheKey: Hashable {
```
Comment (JakeLin) [inline IBAnimatable/PresentationPresenterChacheKey.swift]: Can we directly use `PresentationAnimationType` enum as the key, and let `PresentationAnimationType: Hashable`  and provide ` static func == (lhs: PresentationPresenterChacheKey, rhs: PresentationPresenterChacheKey)`? If we can, then we don't need the additional struct. enum is also a value type.

**Review by JakeLin — CHANGES_REQUESTED** [review]
Can you check whether we can use the enums `PresentationAnimationType` and `TransitionAnimationType` as the key?

**Comment by JakeLin** [comment]
@mmadjer thanks for your PR. Sorry, I have to disappoint you because I think we may be able to use the enums as the key. Can you double check this approach for me?  We also need to rebase since there is some conflict in `TransitionPresenterManager.swift`.

File: IBAnimatable/PresentationPresenterChacheKey.swift (hunk 1 above)
Comment (mmadjer) [reply IBAnimatable/PresentationPresenterChacheKey.swift]: I thought about that but could not find a proper way so far. I will have another look.
I'll give a try and apply the Hashable to the two enums. Why I haven't think about that before. ;o/
;o)

**Comment by mmadjer** [comment]
Ok, I applied the Hashable to the two enums and using it as the key.

File: IBAnimatable/PresentationAnimationType.swift [hunk 2]
```diff
@@ -49,5 +49,15 @@ public enum PresentationAnimationType: IBEnum {
       return nil
     }
   }
+
+}
+
+extension PresentationAnimationType: Hashable {
```
Comment (JakeLin) [inline IBAnimatable/PresentationAnimationType.swift:55]: Perfect 👍

**Review by JakeLin — APPROVED** [review]
Perfect, looks better.

**Comment by JakeLin** [comment]
@mmadjer Thanks for the super quick update, LGTM, can you do a rebase from the master and merge it when you free it is ready.

**Review by JakeLin — APPROVED** [review]
perfect, I always forget about it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,7 +9,8 @@ All notable changes to this project will be documented in this file.
 None
 
 #### Enhancements
-None
+- Use Swift Enum as key in our internal presentation and transition cache.
```
Comment (tbaranes) [inline CHANGELOG.md]: Not sure that we should write a line about this since it's an internal change

**Review by tbaranes — APPROVED** [review]

**Comment by tbaranes** [comment]
Not sure that we should add an entry in the CHANGELOG since it's an internal change. What do you think?

**Comment by JakeLin** [comment]
@tbaranes, IMO we need that. It maybe a breaking change for someone uses the Singleton directly (not 💯 % sure since I am replying this on my mobile and didn't check the source code). That record can help the user find the issue.

**Pull request merged**